File: tests/rosetta/transpiler/php/count-the-coins-2.php
<?php
ini_set('memory_limit', '-1');
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function countChange($amount) {
  $ways = [];
  $i = 0;
  while ($i <= $amount) {
  $ways = array_merge($ways, [0]);
  $i = $i + 1;
};
  $ways[0] = 1;
  foreach ([100, 50, 25, 10, 5, 1] as $coin) {
  $j = $coin;
  while ($j <= $amount) {
  $ways[$j] = $ways[$j] + $ways[$j - $coin];
  $j = $j + 1;
};
};
  return $ways[$amount];
}
$amount = 1000 * 100;
echo rtrim('amount, ways to make change: ' . _str($amount) . ' ' . _str(countChange($amount))), PHP_EOL;

File: tests/rosetta/transpiler/FS/count-the-coins-2.fs
// Generated 2025-07-28 11:14 +0700

exception Return

let rec countChange (amount: int) =
    let mutable __ret : int = Unchecked.defaultof<int>
    let mutable amount = amount
    try
        let mutable ways: int array = [||]
        let mutable i: int = 0
        while i <= amount do
            ways <- Array.append ways [|0|]
            i <- i + 1
        ways.[0] <- 1
        for coin in [|100; 50; 25; 10; 5; 1|] do
            let mutable j = coin
            while (int j) <= amount do
                ways.[j] <- (ways.[j]) + (ways.[j - coin])
                j <- (int j) + 1
        __ret <- ways.[amount]
        raise Return
        __ret
    with
        | Return -> __ret
let amount: int = 1000
printfn "%s" ((("amount, ways to make change: " + (string amount)) + " ") + (string (countChange amount)))
